app/models/petition.rb

```diff
@@ -156,6 +155,13 @@ def self.counts_by_state
     counts_by_state
   end
 
+  def publish!
+    self.state = Petition::OPEN_STATE
+    self.open_at = Time.zone.now
```

Comment:
I blindly copied it from the controller impl. ;)  But you're right, I'll change it and do a sweep through the rest of the app too.